R/utils.R

```diff
@@ -0,0 +1,325 @@
+# Helper functions for do_read_bib()
+
+#' Parse (at)string keys
+#'
+#' Returns a table with all the string values and their value to be used on the
+#' `bib` file.
+#'
+#' @noRd
+parse_strings <- function(map_string,
+                          stringlines) {
+
+  # This needs to be done in a loop, since strings can depend also
+  # on values of previous strings
+
+  parsed_string <- NULL
+
+  for (i in seq_len(length(stringlines))) {
```

Comment:
If `stringlines` is a vector, let's use `seq_along()` 

```suggestion
  for (i in seq_along(stringlines)) {
```